### PR TITLE
feat(web): optimize the state of app

### PR DESCRIPTION
**Pull request by imyuanx**
optimize the state of app when I control app to start or restarting or stop.

## Before

laf-before

## After
laf-after

### CODE REVIEW

**Comment by 0fatal** [comment]
the "phase" field can only be used by backend, try another method:

Pause the application and poll the backend api to request the app state

**Review by 0fatal — CHANGES_REQUESTED** [review]
add "cursor-pointer" to the icon, don't add to CopyText directly

**Comment by imyuanx** [comment]
> add "cursor-pointer" to the icon, don't add to CopyText directly

This PR is about app state

File: web/src/constants/index.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ export const Pages = {
 
 export enum APP_STATUS {
   // [ "Running", "Stopped", "Restarting", "Deleted" ]
+  Starting = "Starting",
```
Comment (newfish-cmyk) [inline web/src/constants/index.ts:19]: remove

File: web/src/pages/home/mods/List/index.tsx [hunk 2]
```diff
@@ -165,6 +165,12 @@ function List(props: { appListQuery: any; setShouldRefetch: any }) {
                           display={"block"}
                           onClick={async (event) => {
                             event?.preventDefault();
+                            updatePhase(
+                              item.appid,
+                              item.phase === APP_STATUS.Stopped
+                                ? APP_STATUS.Starting
```
Comment (newfish-cmyk) [inline web/src/pages/home/mods/List/index.tsx:171]: use "APP_PHASE_STATUS" instead

File: web/src/pages/home/index.tsx [hunk 3]
```diff
@@ -39,6 +42,22 @@ function HomePage() {
     },
   );
 
+  function updatePhase(appid: string, type: keyof typeof APP_PHASE_STATUS) {
+    const newappListQueryData = appListQuery.data?.data.map((item: TApplicationItem) => {
```
Comment (newfish-cmyk) [inline web/src/pages/home/index.tsx:46]: newAppListQueryData

**Review by newfish-cmyk — CHANGES_REQUESTED** [review]

**Comment by imyuanx** [comment]
Done

**Review by newfish-cmyk — APPROVED** [review]

**Review by 0fatal — APPROVED** [review]

**Pull request merged**